docs/utilities/data_classes.md

```diff
@@ -248,6 +250,189 @@ Verify Auth Challenge | `data_classes.cognito_user_pool_event.VerifyAuthChalleng
         do_something_with(user_attributes)
     ```
 
+#### Define Auth Challenge Example
+
+=== "app.py"
+
+    ```python
+    from aws_lambda_powertools.utilities.data_classes.cognito_user_pool_event import DefineAuthChallengeTriggerEvent
+
+    def handler(event, context):
+        _event: DefineAuthChallengeTriggerEvent = DefineAuthChallengeTriggerEvent(event)
+        if (
+            len(_event.request.session) == 1
+            and _event.request.session[0].challenge_name == "SRP_A"
+        ):
+            _event.response.issue_tokens = False
+            _event.response.fail_authentication = False
+            _event.response.challenge_name = "PASSWORD_VERIFIER"
+        elif (
+            len(_event.request.session) == 2
+            and _event.request.session[1].challenge_name == "PASSWORD_VERIFIER"
+            and _event.request.session[1].challenge_result
+        ):
+            _event.response.issue_tokens = False
+            _event.response.fail_authentication = False
+            _event.response.challenge_name = "CUSTOM_CHALLENGE"
+        elif (
+            len(_event.request.session) == 3
+            and _event.request.session[2].challenge_name == "CUSTOM_CHALLENGE"
+            and _event.request.session[2].challenge_result
+        ):
+            _event.response.issue_tokens = True
+            _event.response.fail_authentication = False
+        else:
+            _event.response.issue_tokens = False
+            _event.response.fail_authentication = True
```

Comment:
Can we make this branching logic simpler to read? e.g. separate functions for each challenge, or have class methods to abstract that more easily?

Reply:
@heitorlessa  agreed, but this is based on the AWS code example:
 - https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-define-auth-challenge.html